koku/api/user_settings/views.py

```diff
@@ -56,38 +59,43 @@ def __post_init__(self):
         else:
             raise SettingsInvalidFilterException("Invalid not a user setting")
 
-    def update_cost_type(self, request_data, schema_name):
-        serializer = UserSettingUpdateCostTypeSerializer(schema_name, request_data)
+    def update_cost_type(self, schema_name):
+        serializer = UserSettingUpdateCostTypeSerializer(schema_name, self.request.data)
         if serializer.is_valid(raise_exception=True):
-            set_cost_type(schema_name, request_data.get(self.setting))
+            set_cost_type(schema_name, self.request.data.get(self.setting))
             invalidate_view_cache_for_tenant_and_source_type(schema_name, Provider.PROVIDER_AWS)
             return Response(status=status.HTTP_204_NO_CONTENT)
 
-    def update_currency(self, request_data, schema_name):
-        serializer = UserSettingUpdateCurrencySerializer(schema_name, request_data)
+    def update_currency(self, schema_name):
+        serializer = UserSettingUpdateCurrencySerializer(schema_name, self.request.data)
         if serializer.is_valid(raise_exception=True):
-            set_currency(schema_name, request_data.get(self.setting))
+            set_currency(schema_name, self.request.data.get(self.setting))
             invalidate_view_cache_for_tenant_and_all_source_types(schema_name)
             return Response(status=status.HTTP_204_NO_CONTENT)
 
-    def retrieve_user_settings(self, request):
-        users_setting = {"settings": {self.setting: self.get_param(request)}}
+    def retrieve_user_settings(self):
+        users_setting = {"settings": {self.setting: self.get_param(self.request)}}
         users_setting = UserSettingSerializer(users_setting, many=False).data
         return Response(users_setting)
 
 
 class AccountSettings(APIView):
     """Settings views for all user settings."""
 
-    permission_classes = [permissions.AllowAny]
+    def get_permissions(self):
+        if self.request.method.lower() == "get":
+            permission_classes = [DeprecatedSettingsAccessPermission]
+        else:
+            permission_classes = [SettingsAccessPermission]
+        return [permission() for permission in permission_classes]
 
     def get(self, request, *args, **kwargs):
         """Gets a list of users current settings."""
         if not kwargs:
             user_settings = get_account_settings(request)
         else:
-            param_handler = SettingParamsHandler(kwargs["setting"])
-            user_settings = param_handler.retrieve_user_settings(request).data
+            param_handler = SettingParamsHandler(kwargs["setting"], request)
```

Comment:
Are we guaranteed to have a `settings` key in `kwargs`? There may be a potential for a `KeyError` here.

Reply:
Yes! If kwargs exist, we are guaranteed a setting kwarg because the way the paths are set up. 
```
path("account-settings/", AccountSettings.as_view(), name="account-settings"),
path("account-settings/<str:setting>/", AccountSettings.as_view(), name="get-account-setting"),
```